tensorflow/compiler/jit/xla_compilation_cache.h

```diff
@@ -189,6 +214,30 @@ class XlaCompilationCache : public ResourceBase {
   absl::flat_hash_map<string, ClusterCompileStats> cluster_compile_stats_
       TF_GUARDED_BY(cluster_compile_stats_mu_);
 
+  struct AsyncCompilation {
+    mutex async_compilation_mu_;
+
+    // Number of threads for asynchronous compilations.
+    static constexpr int64 kNumCompilerThreads = 10;
+
+    // Maximum number of ongoing compilations.
+    static constexpr int64 kMaxNumOngoingCompilations = kNumCompilerThreads;
+
+    // Number of ongoing compilations.
+    int64 num_ongoing_compilations GUARDED_BY(async_compilation_mu_) = 0;
+
+    // Pool of threads for asynchronous compilations.
+    std::unique_ptr<thread::ThreadPool> compiler_threads;
```

Comment:
This doesn't really solve the problem right; if someone adds more stuff after `async_compilation_` then those things will still get destroyed before the thread pool right?

I meant doing:

```
 private:
  std::unique_ptr<AsyncCompilation> async_compilation_;

...
  ~ XlaCompilationCache() {
    async_compilation_.reset();
  }
```

Then `async_compilation_` will be destroyed before other fields, even if more fields are added after it.

Reply:
it does solve the problem. Destructors get called in reverse order, and  ~ XlaCompilationCache() gets called first before any member destructor.